### PR TITLE
Update DALI PyTorch ligthing example to work with the newest lighting

### PR DESCRIPTION
**Pull request by JanuszL**
- updates DALI PyTorch lighting example to work with the newest lighting
- moves to the functional API

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a n DALI PyTorch lighting example incompatibility with the latest PyTorch lighting version 

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     add a work around to free GPU booked by the previous instance of PyTorch lighting Trainer class
 - Affected modules and functionalities:
     DALI PyTorch lighting example
 - Key points relevant for the review:
     NA
 - Validation and testing:
     current test apply
 - Documentation (including examples):
     example is updated

**JIRA TASK**: *[DALI-1865]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Review by awolant — APPROVED** [review]

File: docs/examples/frameworks/pytorch/pytorch-lightning.ipynb [hunk 1]
```diff
@@ -184,47 +182,38 @@
...
-    "            output_layout=\"CHW\")\n",
-    "        self.to_int64 = ops.Cast(dtype=types.INT64, device=device)\n",
-    "\n",
-    "    def define_graph(self):\n",
-    "        inputs, labels = self.reader(name=\"Reader\")\n",
-    "        images = self.decode(inputs)\n",
-    "        images = self.cmn(images)\n",
-    "        if self.device == \"gpu\":\n",
+    "def CreateMnistPipeline(batch_size, device, device_id=0, shard_id=0, num_shards=1, num_threads=4, seed=0):\n",
+    "    pipe = Pipeline(batch_size, num_threads, device_id, seed)\n",
```
Comment (klecki) [inline docs/examples/frameworks/pytorch/pytorch-lightning.ipynb:193]: How about using the decorator?

File: docs/examples/frameworks/pytorch/pytorch-lightning.ipynb [hunk 2]
```diff
@@ -325,8 +307,11 @@
     }
    ],
    "source": [
+    "# even if previous Trainer finihsed his work it still keeps the GPU booked, hack to free it\n",
```
Comment (klecki) [inline docs/examples/frameworks/pytorch/pytorch-lightning.ipynb:310]: Just a nitpick, should this really be called a `hack`? Maybe
```suggestion
    "# Even if previous Trainer finished his work it still keeps the GPU booked, force it to release the device.\n",
```

+ there is a typo in `finished`. Same below.

**Review by klecki — APPROVED** [review]
Small nitpick + optional pipeline decorator.

File: docs/examples/frameworks/pytorch/pytorch-lightning.ipynb (hunk 2 above)
Comment (JanuszL) [reply docs/examples/frameworks/pytorch/pytorch-lightning.ipynb:310]: I don't think that changing the env variables set by lighting under the hood can be called anything else than a "hack".

File: docs/examples/frameworks/pytorch/pytorch-lightning.ipynb (hunk 1 above)
Comment (JanuszL) [reply docs/examples/frameworks/pytorch/pytorch-lightning.ipynb]: Done

File: docs/examples/frameworks/pytorch/pytorch-lightning.ipynb (hunk 2 above)
Comment (JanuszL) [reply docs/examples/frameworks/pytorch/pytorch-lightning.ipynb]: Fixed

**Review by klecki — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Pull request merged**